File: Threshold/adaptive.cpp
#include  <opencv2/opencv.hpp>
#include <iostream>

using namespace std;
using namespace cv;

int main(int c,char **argv)
{
	Mat img = imread(argv[1],1);
	
	if(img.empty())
	{
		cout<<"Error,image can't be read"<<endl;
		return 0;
	}
	Mat gray,thresh_mean,thresh_gaussian;
	cvtColor(img,gray,CV_BGR2GRAY);
	adaptiveThreshold(gray,thresh_mean,255,CV_ADAPTIVE_THRESH_MEAN_C,CV_THRESH_BINARY_INV,17,12);
	adaptiveThreshold(gray,thresh_gaussian,255,CV_ADAPTIVE_THRESH_GAUSSIAN_C,CV_THRESH_BINARY_INV,17,12 );
	
	namedWindow("thresh_gaussian",WINDOW_NORMAL);
	imshow("thresh_gaussian",thresh_gaussian);
	namedWindow("thresh_mean",WINDOW_NORMAL);
	imshow("thresh_mean",thresh_mean);
	waitKey(0);
}

File: Threshold/adaptive.py
import cv2 
import numpy as np
import sys

image = cv2	.imread(sys.argv[1],0)

th2 = cv2.adaptiveThreshold(image,255,cv2.ADAPTIVE_THRESH_MEAN_C,\
            cv2.THRESH_BINARY_INV,17,12)
th3 = cv2.adaptiveThreshold(image,255,cv2.ADAPTIVE_THRESH_GAUSSIAN_C,\
            cv2.THRESH_BINARY_INV,17,12)

cv2.namedWindow("th2",cv2.WINDOW_NORMAL)
cv2.imshow("th2",th2)
cv2.namedWindow("th3",cv2.WINDOW_NORMAL)
cv2.imshow("th3",th3)
cv2.waitKey(0)